Add logging

wydra98 opened the pull request:
* Add example ZIO.logInfo in some methods (I chose methods that create something)

Example message:
![image](https://github.com/softwaremill/realworld-tapir-zio/assets/50871708/60d534c8-5dd9-4a60-8531-c5f10f2aeef5)

adamw commented:
Let's use logback as the logging backend (see adopt-tapir for how to configure in ZIO)

wydra98 commented:
@adamw Sorry, but I think I told you wrong earlier. Logback is available in the project and is functioning correctly. The only thing I want to add is a few examples of using `ZIO.logInfo`. I understand from the issue that this is what's missing.

adamw commented:
I can see that slf4j is configured in ZIO, but there's no logback.xml in https://github.com/softwaremill/realworld-tapir-zio/tree/master/src/main/resources ? The log output seems weird, not something I would expect from a normal logback-based application

adamw commented on src/main/scala/com/softwaremill/realworld/users/UsersService.scala:
```diff
@@ -29,14 +29,15 @@ class UsersService(authService: AuthService, usersRepository: UsersRepository):
         } yield UserResponse(userWithToken(emailClean, usernameClean, jwt))
       }
-    } yield user
+      _ <- ZIO.logInfo(s"Successful register of user ${userResponse.user}")
```
[english] `Successfully registered user ...` I think is (more ;) ) correct

adamw commented:
Also, if we are adding logging, let's add it consistently, for all paths which lead to state changes in the system - not just as an example. Probably in the services?

wydra98 commented:
@adamw ok, i added logback.xml file, but i see that it doesn't change a lot. Default ZIO pattern is almost the same as in this file. Besides, I also added logging for all paths which lead to state changes in the system like you wrote.

adamw commented:
Well ... do the log messages follow the pattern as defined in logback.xml?

wydra98 commented:
Yes, the log messages follow the pattern as defined in the logback.xml file

adamw commented on src/main/scala/com/softwaremill/realworld/articles/comments/CommentsService.scala:
```diff
@@ -40,13 +42,16 @@ class CommentsService(
     articlesRepository.findArticleIdBySlug(slug).someOrFail(NotFound(ArticleNotFoundMessage(slug)))
 
 object CommentsService:
+  private val ArticleAndAuthorIdsNotFoundMessage: CommentId => String = (commentId: CommentId) =>
```
1. why not `def`s?
2. are these comments reused somehow? if not, why not simply inline them to the logInfo invocations?

wydra98 replied:
1. Generally, I relied on the code that already existed, which used def instead of val, but indeed, that needs to be changed.
2. I wanted to separate these two things and have the code written in a consistent style, but that probably needs to be changed as well.

adamw replied:
For such a simple thing as log messages let's just write the strings directly
